### DISCUSSION TITLE
Bagaimanakah mengurutkan kolom dg tipe data character pada visualisasi Line Plot ?

### DISCUSSION THREAD

**Discussion by Micoalgo** [teamalgoritma/community#542] [Q&A] [answered]
Dari data yg di olah pada kolom Periode yg merupakan gabungan bulan_tagihan dan tahun_tagihan, sbb ;
<img width="556" alt="image" src="https://user-images.githubusercontent.com/123436963/218377433-f72205ed-e3d2-458a-83a3-436f6a034d32.png">

<img width="596" alt="image" src="https://user-images.githubusercontent.com/123436963/218376655-4667ede5-a16f-4459-8e72-2a89c44b4716.png">

Ketika di buatkan Line Plot nya, muncul tdk dimulai dg bulan 3/2021 tetapi 1/2022, 1/2023 dst yaitu mengikuti urutan angka bulannya saja, sbb;
<img width="415" alt="image" src="https://user-images.githubusercontent.com/123436963/218377517-35eef13d-3075-47af-9528-fdec3e55e934.png">

<img width="623" alt="image" src="https://user-images.githubusercontent.com/123436963/218376909-e92d32fc-74cf-432e-af79-034e680bec1e.png">
 
Bagaimanakah caranya agar line plot tsb scr berurut dimulai dari 3/2021, 4/2021 dst ? Terimakasih

**Comment by nadyadtm** [comment, marked as answer]
Dear Pak @Micoalgo 

Izin menjawab, bila dalam data tersebut hanya ada informasi bulan dan tahun, bapak dapat menggunakan library ``lubridate`` dengan menggunakan fungsi ``my()`` untuk konversi tipe data dari char ke date agar dapat terurut secara otomatis nanti di plotnya.

````
tspermonth$converted_periode <- my(tspermonth$periode)
````

Kemudian, agar seluruh bulan dan tahun terlihat di grafik dan untuk menyesuaikan format bulan/tahunnya, bapak dapat menambahkan lagi ``scale_date_x``, sehingga nantinya code bapak akan menjadi sebagai berikut

````
ggplot(data = tspermonth, aes(x=converted_periode, y=total_tagihan)) + geom_line(group=1) + theme(axis.text, x = element_text(angle = 90)) + scale_x_date(breaks = "1 month", date_labels="%m/%Y")
````

Semoga membantu

**Comment by Micoalgo** [comment]
Terimakasih jawabannya
